### PR TITLE
module putaway apply

### PR DESCRIPTION
**Pull request by mylbco**
Standard Odoo can use only a product putaway strategy on product_category. In many cases this is not sufficient and warehouse manager wants to be able to use a putaway strategy on product level.

selection_700

selection_701

### CODE REVIEW

**Comment by gurneyalex** [comment]
@mylbco several flake8 issues to take care of: https://travis-ci.org/OCA/stock-logistics-workflow/jobs/66230098#L163

File: putaway_apply/__init__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>
+#
+##############################################################################
+import putaway_apply
```
Comment (gurneyalex) [inline putaway_apply/__init__.py]: from . import putaway_apply

File: putaway_apply/security/stock_security.xml [hunk 2]
```diff
@@ -0,0 +1,119 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+<data noupdate="0">
+
+    <record id="group_locations" model="res.groups">
+        <field name="name">Manage Multiple Locations and Warehouses</field>
+        <field name="category_id" ref="base.module_category_hidden"/>
+     </record>
```
Comment (gurneyalex) [inline putaway_apply/security/stock_security.xml]: is this meant to be the same as `stock.group_locations`?

You seem th be doing strange things and recreating groups here and below, where you would normally want to modify existing groups (which requires using the qualified xmlid).

File: putaway_apply/security/stock_security.xml [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+</data>
+<data noupdate="1">
+<!-- multi -->
+
+    <record model="ir.rule" id="stock_picking_rule">
+        <field name="name">stock_picking multi-company</field>
+        <field name="model_id" search="[('model','=','stock.picking')]" model="ir.model"/>
+        <field name="global" eval="True"/>
+        <field name="domain_force">['|',('company_id','=',False),('company_id','child_of',[user.company_id.id])]</field>
+    </record>
```
Comment (gurneyalex) [inline putaway_apply/security/stock_security.xml]: these rules seem to be unrelated to the module. Am I missing something?

File: putaway_apply/putaway_apply_view.xml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                       		</field>
+                       	</page>
+            		</notebook>
+                </xpath>
+            </field>
+        </record>
+
+
+	</data>
+</openerp>
```
Comment (gurneyalex) [inline putaway_apply/putaway_apply_view.xml]: please fix the indentation of this file

File: putaway_apply/security/stock_security.xml [hunk 5]
```diff
@@ -0,0 +1,119 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (gurneyalex) [inline putaway_apply/security/stock_security.xml]: this file is not listed in `__openerp__.py` -> it should probably be completely removed from the PR.

File: putaway_apply/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>
+#
+##############################################################################
+{
+    "name" : "Putaway apply",
+    "version" : "0.1",
+    "author": "Bubbles-iT, Odoo Community Association (OCA).",
+    "category": 'Warehouse Management',
+    "description":' Usually Putaway strategy works on categories matched to the product on Shipment. This module does it on Product level.',
```
Comment (gurneyalex) [inline putaway_apply/__openerp__.py]: can you move out the description to a `README.rst` file? 

Here's a template: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: putaway_apply/__openerp__.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    "website": "http://www.bubbles-it.be",
+    "depends" : ["stock"],
+    "demo" : [
+             ],
```
Comment (gurneyalex) [inline putaway_apply/__openerp__.py]: remove the unused key.

**Comment by gurneyalex** [comment]
module should be renamed. I suggest  `stock_product_putaway_strategy`

Not sure if this belongs to OCA/stock-logistics-workflow or OCA/stock-logistics-warehouse

**Comment by mylbco** [comment]
ok Alexandre, I know what to do !

Thx,

P

Peter Langenberg
Zaakvoerder

Bezoek onze website :  http://goog_220292044/www.bubbles-it.be
Volg op twitter via : @mylbco
Adres : Luikstraat 67, 9160 Lokeren
https://www.google.be/maps/place/Luikstraat,+9160+Lokeren

 “If you think a professional is expensive, wait till you try an amateur.”

Paul "Red" Adair

2015-06-15 11:23 GMT+02:00 Alexandre Fayolle notifications@github.com:

**Comment by gurneyalex** [comment]
and an automated test would be very welcome.

File: putaway_apply/security/stock_security.xml (hunk 5 above)
Comment (mylbco) [inline putaway_apply/security/stock_security.xml]: Yes True, I'll remove it.

**Comment by Cedric-Pigeon** [comment]
@mylbco Hello, do you plan to achieve this PR ? Otherwise it will be closed in 3 days. Thanks.

**Comment by pedrobaeza** [comment]
No answer, so I close. Feel free to reopen if needed. Note that in later versions there are already a module stock_putaway_product for the same purpose.
